src/lib/NodeSmartLinkProvider.ts

```diff
@@ -176,14 +176,25 @@ export class NodeSmartLinkProvider {
   private onDomMutation = (mutations: MutationRecord[]): void => {
     const attrName = dataToDatasetAttributeName(DataAttribute.ElementCodename);
 
-    const relevantMutations = mutations.filter(
-      (mutation) =>
+    const relevantMutations = mutations.filter((mutation: MutationRecord) => {
+      const isTypeRelevant = mutation.type === 'childList';
+      const isTargetRelevant =
         mutation.target instanceof Element &&
-        mutation.type === 'childList' &&
-        ![HighlighterElementTag, HighlighterContainerTag].includes(mutation.target.tagName) &&
-        Array.from(mutation.addedNodes).some((node) => (node as HTMLElement).tagName !== HighlighterElementTag) &&
-        Array.from(mutation.removedNodes).some((node) => (node as HTMLElement).tagName !== HighlighterElementTag)
-    );
+        ![HighlighterElementTag, HighlighterContainerTag].includes(mutation.target.tagName);
+
+      if (!isTypeRelevant && !isTargetRelevant) {
```

Comment:
I believe that a `!isTypeRelevant || !isTargetRelevant` condition would be more appropriate here, as you will be able to filter out more non-relevant cases.

Reply:
You are right, thanks :)